apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -225,7 +225,7 @@ public void AnnotateSpan(Dictionary<string, string> labels)
         {
             GaxPreconditions.CheckNotNull(labels, nameof(labels));
 
-            var currentStack = TraceStack;
+            var currentStack = GetUpdatedStack(TraceStack);
             CheckStackNotEmpty(currentStack);
```

Comment:
Probably best to re-assign `currentStack` to `TraceStack` at some point here so you only do the work of skipped over a particular disposed span once at most.